btr-web/btr-common-components/components/bcros/inputs/PhoneNumber/index.vue

```diff
@@ -0,0 +1,75 @@
+<template>
+  <div class="flex flex-row gap-4">
+    <UFormGroup :name="name + '.countryCode'" class="w-1/4">
+      <BcrosInputsPhoneNumberCountryCode
+        v-model:country-calling-code="phoneNumber.countryCallingCode"
+        v-model:country-code2letter-iso="phoneNumber.countryCode2letterIso"
+        data-cy="phoneNumber.countryCode"
+        :placeholder="$t('placeholder.phoneNumber.countryCode')"
+      />
+    </UFormGroup>
+    <UFormGroup :name="name + '.number'" class="w-1/2">
+      <template #help>
+        <BcrosTooltip
+          :popper="{
+            placement: 'bottom',
+            arrow: true,
+            resize: true
+          }"
+        >
+          <template #tooltip-text>
+            <span class="whitespace-normal place-content: center">
+              {{ $t('helpTexts.phoneNumber.mustBe10DigitsLongTooltip') }}
+            </span>
+          </template>
+          <span class="text-xs">
+            {{ $t('helpTexts.phoneNumber.mustBe10DigitsLong') }}
+          </span>
+        </BcrosTooltip>
+      </template>
+      <UInput
+        v-model="maskedPhoneNumber"
+        v-maska:unmaskedvalue.unmasked
+        variant="bcGov"
+        data-cy="phoneNumber.number"
+        :data-maska="inputMask"
+        :placeholder="$t('placeholder.phoneNumber.number')"
+      />
+    </UFormGroup>
+    <UFormGroup :name="name + '.extension'" class="w-1/4">
+      <UInput
+        v-model="phoneNumber.extension"
+        :placeholder="$t('placeholder.phoneNumber.extension') + ' (' + $t('general.optional') + ')'"
+        variant="bcGov"
+        data-cy="phoneNumber.extensionCode"
+      />
+    </UFormGroup>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { vMaska } from 'maska/vue'
+import { watch } from 'vue'
+import { PhoneSchemaType } from '~/interfaces/zod-schemas-t'
+
+const phoneNumber = defineModel<PhoneSchemaType>({ required: true })
+
+defineProps({
+  name: { type: String, default: 'phoneNumber' }
+})
+
+const northAmericaMask = '(###) ###-####'
+const otherMask = '##############'
```

Comment:
I don't think we need to mask it in the UI since the api will have to do this as well anyways?

Reply:
We use mask just for display purposes and store number in string format with only numbers.

Reply:
- duplicates, there should be no any if you see duplicates can you list them, so I can clean them up.
- there are few other countries in Caribbean with +1 as country code.
- i created custom search function to filter the options. before it was required to escape the plus sign as it was regex search. Now it is just plain inclusive string.

Reply:
I'm not going to go through the whole list haha but you can see examples of duplicates above

Reply:
The masking has to be from the api otherwise anyone could see the info from the network tab

Reply:
and its still not adding phone number to the submission fyi

Reply:
```
{
    countryCallingCode: "262"
    countryCode2letterIso: "RE"
    countryNameEn: "Réunion"
    countryNameLocal: "La Réunion"
    label: "+262"
}

{
    countryCallingCode: "262"
    countryCode2letterIso: "YT"
    countryNameEn: "Mayotte"
    countryNameLocal: "Mayotte"
    label: "+262"
}
```
If you see from this, they are not duplicates.
Also, I readded tooltip for the flag in the options so they know for what country the calling code is being selected.

And I think the "mask" stuff got misunderstood, I was thinking about the 'input-mask' which is 
having brackets, and dashes and everything, not masking the input in regards to being obscured as data protection. As this was not implemented in this ticket.

Reply:
And yes, readded it to the converters.

Reply:
can we collapse the ones with the same flag and assume they follow the same format rules?

Reply:
If I collapse them, what country do I select as the phone country ?

For the tooltip, I used the local name of the country. Do they break the font ? I checked few on my PC and they looked good.